app/models/vacancy.rb

```diff
@@ -209,19 +205,21 @@ def can_receive_job_applications?
   end
 
   def allow_key_stages?
-    allowed_phases = %w[primary middle secondary through]
+    allowed_phases = %w[primary secondary through]
     allowed_roles = %w[teacher headteacher deputy_headteacher assistant_headteacher
                        head_of_year_or_phase head_of_department_or_curriculum teaching_assistant]
 
     phases.intersect?(allowed_phases) && job_roles.intersect?(allowed_roles)
   end
 
   def allow_phase_to_be_set?
-    school_phases.none?
+    school_phases = organisations.schools.filter_map(&:phase).uniq
+
+    !(school_phases.intersect? %w[nursery primary secondary sixth_form_or_college])
```

Comment:
This list is the same as the others - maybe good to have a test to show that we now prompt for phase in these cases